aten/src/ATen/core/dispatch/Dispatcher.h

```diff
@@ -350,15 +382,30 @@ inline Return Dispatcher::redispatch(const TypedOperatorHandle<Return (Args...)>
       DispatchKeySet(DispatchKeySet::FULL_AFTER, currentDispatchKey),
       args...
     );
-  return callWithDispatchKey<Return, Args...>(op, dispatchKey, args...);
+  // do not use RecordFunction on redispatch
+  return callWithDispatchKey<Return, Args...>(op, dispatchKey, false, args...);
 }
 
 inline void Dispatcher::callBoxed(const OperatorHandle& op, Stack* stack) const {
   // note: this doesn't need the mutex because write operations on the list keep iterators intact.
   const auto& entry = op.operatorIterator_->op;
   auto dispatchKey = entry.dispatchKeyExtractor().getDispatchKeyBoxed(stack);
   const auto& kernel = entry.lookup(dispatchKey);
-  kernel.callBoxed(op, stack);
+
+  if (entry.isObserved()) {
```

Comment:
Note: the absence of `useRecordFunction` here introduces an API inconsistency between `callBoxed` and `callWithDispatchKey` - one lets you control RecordFunction and the other doesn't. If we go with the `useRecordFunction` approach, I'm thinking we should add it to `callBoxed` as well (we can avoid callsite churn by defaulting it). 

But also, if you're inclined to dislike call-site-decides because of leakage, here's some more :P

Reply:
I was wondering in what cases callBoxed would be called with BackendSelect, or is it for the symmetry

Reply:
Yeah, just for the symmetry. Unless there's some history that I'm missing, there's no reason to think we don't want the same knobs on boxed calls as unboxed. 

Note: I wouldn't necessarily burden your PR with the addition - I just wanted to note it in the context of the discussion about filtering BackendSelect et al.